### ISSUE TITLE
Mac: problem on startup when splashscreen is enabled

### ISSUE THREAD

**Issue by dliessi** [wbsoft/frescobaldi#428]
Reported by @philmassart (Mac OS 10.7 and 10.9):

> [...] when checking the option "Show splashscreen", Frescobaldi doesn't show its window; I have to click on the Frescobaldi icon in the dock. I don't need to do that when the splashscreen doesn't show.

... by @PaulMorris (Mac OS 10.9):

> I am also seeing the splash-screen-then-no-new-window-until-click-dock-icon behavior, [...].  I find that this problem doesn't occur if I uncheck "Show Splash Screen on Startup" in preferences, [...].

... and by three other users (Mac OS 10.9):

> After the splash screen the application moves to the background. I have to click on the Frescobaldi icon in the dock to bring it to the front. Using [command]+[tab] to cycle through active applications doesn’t bring it to the front unless I’ve already clicked on the icon in the dock.

@philmassart: do you see this both in 10.9 and 10.7 or only on 10.9?
I suspect that the problem might be specific to 10.9.

**Comment by dliessi** [comment]
Issue #603:

> Reported by a user:
>
>> on Mac, when I launch Frescobaldi it does not show the document. I always have to go to " Window > New Window" to see the lilypond code and the music. Is there a workaround?
>
> There was a reply by @philmassart :
>
>> That bug is found since some time now (and is linked, I think, to the mac implementation of the QT4 framework). The only other way now is to click once more on the Frescobaldi icon in the Dock since Frescobaldi is running
>
> Can it be a Frescobaldi bug? It seems Frescobaldi is started but does not show or create a main window...

**Comment by dliessi** [comment]
> Can it be a Frescobaldi bug? It seems Frescobaldi is started but does not show or create a main window...

@wbsoft It seems to be related to the splash screen.
The problem doesn't occur on 10.6; I'll try on 10.10 as soon as I can.

**Comment by PaulMorris** [comment]
I just checked on 10.10 and the problem occurs there too.  When the "show splash screen" preference is checked, no window is opened after the splash screen goes away.  When "show splash screen" is unchecked, the window opens as expected.

**Comment by dliessi** [comment]
[Reported by Steve Noland](http://lists.gnu.org/archive/html/lilypond-user/2015-03/msg00401.html):
> Also, for the developers, it appears that the pre-build .dmg version of
> Frescobaldi opens correctly with the full set of windows, but the version
> built from MacPorts will launch and show the splash, but one has to click
> the dock icon again to get the full window set.

@PaulMorris @philmassart can you please test if indeed the pre-built application is not affected by this issue?

**Comment by PeterBjuhr** [comment]
Frescobaldi: 2.18 from dmg
Darwin-13.4.0-x86_64-i386-32bit
OS X 10.9.5

The behavior is exactly as described. With a preference for the splash-screen the mainwindow doesn't show, and when unselecting the preference the mainwindow does show.

**Comment by PaulMorris** [comment]
> @PaulMorris <https://github.com/PaulMorris> @philmassart <https://github.com/philmassart> can you please test if indeed the pre-built application is not affected by this issue?
> 
I'm still getting this issue with the pre-built version.  With the splash screen on, no window appears until you click the dock icon.  Without the splash screen, all is well.  

Frescobaldi: 2.18

Python: 2.7.9
Qt: 4.8.6
PyQt4: 4.11.3
sip: 4.16.6
python-ly: 0.9.1
poppler: 0.31.0
python-poppler-qt4: 0.24.0

Operating System:
Darwin-14.1.0-x86_64-i386-64bit

**Comment by AllanKinnaird** [comment]
I can confirm that I am seeing this behaviour on OS X 10.10.3, with Frescobaldi installed from Homebrew. With the splash screen enabled the initial window does not appear when the splash screen closes. Command-tabbing causes the program to disappear into the background (its menu bar does not return on command-tabbing back to the Frescobaldi icon) and only clicking on the dock icon restores it and its main window. After that its behaviour seems normal. Behaviour appears normal from the start if the splash window is disabled.

**Comment by dliessi** [comment]
A strange thing is that on my system (Mac OS 10.6.8) if I disable the splash screen then Frescobaldi automatically opens [the `frescobaldi` script](https://github.com/wbsoft/frescobaldi/blob/master/frescobaldi) in the editor when it starts.

I'm trying to understand why this happens and how to solve it, but I really cannot understand the difference in behaviour with respect to later versions of Mac OS X.

**Comment by kierenmacmillan** [comment]
I think this is the same problem as I've been fighting ever since I began using Frescobaldi (about 6 months ago): None of the standard Mac OS X application-shifting functions work as expected. "Hide Frescobaldi" doesn't, nor does Frescobaldi hide when another application triggers "Hide Others", nor does Frescobaldi appear when using Command-Tab to switch applications. [Mac OS X 10.9.5]

**Comment by fishnet37222** [comment]
I can confirm that this issue is still occurring on OS X 10.11.5.

**Comment by kierenmacmillan** [comment]
I tried the suggested fix — a year ago — and the problem persists.

**Comment by dliessi** [comment]
reported by a user (see #863):
> Frescobaldi: 2.19.0 -- Python: 2.7.11 -- python-ly: 0.9.4 -- Qt: 4.8.7 -- PyQt: 4.11.4 -- sip: 4.17 -- poppler: 0.39.0 -- python-poppler-qt: 0.24.0 -- OS: Darwin-15.6.0-x86_64-i386-64bit
>
> Since several iterations of Frescobaldi the first launching of the app results in app running, but not showing main window. One has to select something in the menu bar, and then the main window will be shown.

**Comment by wbsoft** [comment]
I really need a mac machine to be able to debug these issues....

**Comment by rpspringuel** [comment]
Frescobaldi 3.1, Mac 10.15.2 is still showing this issue.

**Comment by fedelibre** [comment]
Can anybody confirm if this problem is still present in version 4?

**Comment by fedelibre** [comment]
@JunTamura1959 Can you please try enabling the splashscreen in the General Preferences and advise if you can reproduce this bug?

**Comment by JunTamura1959** [comment]
@fedelibre 
No, I don't see this problem anymore on the 4.0.2 app package, or 4.0.3.dev1 launched from the command line. I used to encounter this problem on 3.x.y, but I think it has been resolved in 4.0.x.

Frescobaldi: 4.0.2
Extension API: 0.9.0
Python: 3.12.7
python-ly: 0.9.9
Qt: 6.9.0
PyQt: 6.9.0
qpageview: 1.0.0
OS: macOS-15.5-x86_64-i386-64bit
installation kind: standalone .app bundle

Frescobaldi: 4.0.3.dev1
Extension API: 0.9.0
Python: 3.12.8
Git branch: master
on commit: b5fc043e
python-ly: 0.9.9
Qt: 6.8.1
PyQt: 6.8.0
qpageview: 1.0.0
OS: macOS-15.5-x86_64-i386-64bit
installation kind: command line

**Comment by fedelibre** [comment]
Thanks for confirming. I guess that the analysis by Jean was correct
